### PR TITLE
Redux functionality

### PR DESCRIPTION
**Pull request by Josphat205**
## Process followed

- Used useSelector to consume the state.
- Imported Redux useDispatch
- Set the data inputs in the local React state
- Dispatched a corresponding action

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -13,19 +13,19 @@ function Books() {
     <Container>
       <BooksList>
         {
-          books.map((book)=>{
+          books.map((item)=>{
             return(
-              <Book id={book.id}>
+              <Book id={item.id}>
```
Comment (omar-muhamad) [inline src/components/Books.js]: - [ ] Please give every Book you render a unique key property to make react keep track of every change made. see this [link ](https://reactjs.org/docs/lists-and-keys.html#keys)for more info.

image

File: src/components/Categories.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import styled from 'styled-components';
-
+import {useDispatch} from 'react-redux'
 function Categories() {
+  const dispatch = useDispatch()
   return (
     <Container>
       <h1>Categories will go here</h1>
+      <Button type="button" onClick={()=>dispatch()}>Update Progress</Button>
```
Comment (omar-muhamad) [inline src/components/Categories.js]: - [ ] When you click on the update progress button it supposes to dispatch CHECK_STATUS action and `under construction`  appear on the page but that's not happening. So please fix that.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Josphat205,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Josphat205, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**